source/common/http/conn_manager_impl.cc

```diff
@@ -1088,30 +1088,6 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
     // Allow non websocket requests to go through websocket enabled routes.
   }
 
-  if (hasCachedRoute()) {
```

Comment:
One of the differences I see here is that before this change, the timer was created after lines 1075-1089, which IIUC could send a local reply and not create the timer. In the new code the timer could be created (as part of the call from `refreshCachedRoute()` to `setRoute()`) and then a local-reply will be sent.
What will happen in this case if a timeout occurs?

Reply:
Actually, a default `stream_idle_container_` was already created in the `ActiveStream` constructor. So the possibility of timing out during a local reply already existed.